src/libraries/System.Linq/src/System/Linq/OrderedEnumerable.cs

```diff
@@ -343,6 +376,9 @@ internal override int CompareAnyKeys(int index1, int index2)
         protected override void QuickSort(int[] keys, int lo, int hi) =>
             new Span<int>(keys, lo, hi - lo + 1).Sort(CompareAnyKeys);
 
+        protected override void QuickSortElements(TElement[] element, int lo, int hi) =>
+            new Span<TKey>(_keys, lo, hi - lo + 1).Sort(new Span<TElement>(element, lo, hi-lo +1), _comparer);
```

Comment:
This does not seem to be taking descending sorts into account.

Reply:
Thanks for the catch, I tried using the overload taking Comparison\<TKey> and made a function like CompareAnyKeys. This is causing a huge perf degredation, it appears that it is due to the additional method call to the Comparable\<TKey> created in Sort and and the call to new new Comparison\<TKey>. Since Sort doesn't support descending the only way to maintain this Perf is to restrict this to only when doing ascending which I think is a bad idea. If you don't have any further suggestion I'll abandon this PR.
|      Method |        Job |               Toolchain |   N |        Mean |       Error |      StdDev |      Median |         Min |         Max |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|------------ |----------- |------------------------ |---- |------------:|------------:|------------:|------------:|------------:|------------:|-------:|------:|------:|----------:|
|  LongToList | Job-KMOQSK |       IComparison\<TKey> | 200 | 15,966.5 ns |   736.64 ns |   788.20 ns | 15,986.1 ns | 14,693.0 ns | 17,493.0 ns | 1.1421 |     - |     - |    5104 B |
|  LongToList | Job-BJFYTT |                upstream | 200 | 15,546.1 ns |   773.05 ns |   859.24 ns | 15,410.0 ns | 14,292.5 ns | 17,462.7 ns | 1.3152 |     - |     - |    5904 B |
|  LongToList | Job-BCHDBG |        Comparable\<TKey> | 200 |  3,087.8 ns |   177.38 ns |   189.79 ns |  3,065.8 ns |  2,846.0 ns |  3,512.2 ns | 1.1559 |     - |     - |    5016 B |